src/ai/backend/agent/task_runner/heartbeat.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from collections.abc import Awaitable, Sequence
+from typing import Callable, Optional, override
+
+from ai.backend.common.events.dispatcher import EventProducer
+from ai.backend.common.events.event_types.agent.anycast import (
+    AgentStatusHeartbeat,
+    ContainerStatusData,
+)
+from ai.backend.common.task_runner.types import AbstractTask
+from ai.backend.common.types import AgentId, ContainerStatus, KernelId
+from ai.backend.logging.utils import BraceStyleAdapter
+
+from ..types import (
+    Container,
+)
+
+log = BraceStyleAdapter(logging.getLogger(__spec__.name))
+
+
+class HeartbeatTask(AbstractTask):
+    def __init__(
+        self,
+        agent_id: AgentId,
+        container_enumerator: Callable[
+            [frozenset[ContainerStatus]], Awaitable[Sequence[tuple[KernelId, Container]]]
+        ],
+        event_producer: EventProducer,
+    ) -> None:
+        self._agent_id = agent_id
+        self._container_enumerator = container_enumerator
```

Comment:
I want to avoid receiving callables as arguments as much as possible because it's difficult to track them easily when you pass a callable.